lib/librarian/puppet/source/git.rb

```diff
@@ -37,6 +37,11 @@ def cache!
             raise Error, "Could not checkout #{uri}#{" at #{sha}" unless sha.nil?}: #{e}"
           end
 
+          if environment.config_db["install.strip-dot-git"] == "1"
+            dot_git = filesystem_path.join(".git")
+            dot_git.rmtree if dot_git.directory?
+          end
```

Comment:
question: if you, say, upgrade, does it actually work? I'm not certain how librarian-puppet knows what version exists in the modules directory without the .git directory, so I'd want to make sure that it's not re-cloning the upstream repo every time etc.